## 02-dummy-defaults.patch by Christian Welzel <gawain@camlann.de>
##
## DP: provide some default values for Debian systems

--- typo3-src-4.5.8+dfsg1.orig/dummy/typo3conf/localconf.php
+++ typo3-src-4.5.8+dfsg1/dummy/typo3conf/localconf.php
@@ -8,10 +8,145 @@
 	// Default password is "joh316" :
 $TYPO3_CONF_VARS['BE']['installToolPassword'] = 'bacb98acf97e0b6112b1d1b650b84971';
 
-$TYPO3_CONF_VARS['EXT']['extList'] = 'info,perm,func,filelist,about,version,tsconfig_help,context_help,extra_page_cm_options,impexp,sys_note,tstemplate,tstemplate_ceditor,tstemplate_info,tstemplate_objbrowser,tstemplate_analyzer,func_wizards,wizard_crpages,wizard_sortpages,lowlevel,install,belog,beuser,aboutmodules,setup,taskcenter,info_pagetsconfig,viewpage,rtehtmlarea,css_styled_content,t3skin,t3editor,reports,felogin';
+$TYPO3_CONF_VARS['EXT']['extList'] = 'cms,lang,sv,em,fal,list,tsconfig_help,context_help,extra_page_cm_options,impexp,sys_note,tstemplate,tstemplate_ceditor,tstemplate_info,tstemplate_objbrowser,tstemplate_analyzer,func_wizards,wizard_crpages,wizard_sortpages,lowlevel,install,belog,beuser,aboutmodules,setup,taskcenter,info_pagetsconfig,viewpage,rtehtmlarea,css_styled_content,t3skin,t3editor,reports,info,perm,func,filelist,about,cshmanual,opendocs,recycler,scheduler,linkvalidator,pagetree';
 
 $typo_db_extTableDef_script = 'extTables.php';
 
+$debian_config = '/etc/typo3-dummy/debian-db.php';
+if(is_file($debian_config) && is_readable($debian_config)) {
+  include_once($debian_config);
+
+  $typo_db_username = $dbuser;
+  $typo_db_password = $dbpass;
+  $typo_db = $dbname;
+  $typo_db_host = $dbserver;
+  if(empty($typo_db_host)) $typo_db_host = 'localhost';
+}
+
+// Use the memcached server for speeding up TYPO3, set next line to '1' to enable the framework
+$TYPO3_CONF_VARS['SYS']['useCachingFramework'] = '0';
+$TYPO3_CONF_VARS['SYS']['caching']['cacheBackendAssignments'] = array(
+    'cache_hash' => array(
+        'backend' => 't3lib_cache_backend_File',
+        'options' => array(
+        )
+    ),
+    'cache_pages' => array(
+        'backend' => 't3lib_cache_backend_Memcached',
+        'options' => array(
+            'servers' => array('localhost:11211'),
+        )
+    ),
+    'cache_pagesection' => array(
+        'backend' => 't3lib_cache_backend_Memcached',
+        'options' => array(
+            'servers' => array('localhost:11211'),
+        )
+#        'backend' => 't3lib_cache_backend_Db',
+#        'options' => array(
+#            'cacheTable' => 'cache_pagesection',
+#        )
+    )
+);
+
 ## INSTALL SCRIPT EDIT POINT TOKEN - all lines after this points may be changed by the install script!
 
+// encryption key. set is to a long unique string.
+$TYPO3_CONF_VARS['SYS']['encryptionKey'] = '###ENCKEY###';
+
+// If safe_mode is activated with TYPO3, disable use of
+// external programs
+// typo3-dummy provides links to the most important external programs in
+// /var/lib/typo3-dummy/execdir, so exec() function is enabled by default.
+$TYPO3_CONF_VARS["BE"]["disable_exec_function"] = '0';
+
+// TYPO3 prefers the use of GIF-files and most likely your visitors on
+// your website does too as not all browsers support PNG yet.
+$TYPO3_CONF_VARS["GFX"]["gdlib_png"] = '1';
+
+// set this true to get some better results in GIFBUILDER
+$TYPO3_CONF_VARS['GFX']['png_truecolor'] = '1';
+
+// enabling the use of gdblib2 for image processing
+$TYPO3_CONF_VARS["GFX"]["gdlib_2"] = '1';
+
+// last versions of imagemagick renamed combine to composite. It will
+// be set by basic configuration automatically.
+$TYPO3_CONF_VARS["GFX"]["im_combine_filename"] = 'combine';
+
+// This value should be set to 1 if imagemagick version is greater
+// than 5.2
+$TYPO3_CONF_VARS["GFX"]["im_negate_mask"] = '1';
+
+$TYPO3_CONF_VARS["GFX"]["im_imvMaskState"] = '1';
+
+$TYPO3_CONF_VARS["GFX"]["im_mask_temp_ext_gif"] = '1';
+
+// The value should be 0 if the version of imagemagick is greater than
+// 5, otherwise the creation of effects is getting too slow
+$TYPO3_CONF_VARS["GFX"]["im_no_effects"] = '1';
+
+$TYPO3_CONF_VARS["GFX"]["im_v5effects"] = '1';
+
+// Path to the imagemagick manipulation tools like convert,
+// composite and identify
+$TYPO3_CONF_VARS["GFX"]["im_path"] = '/var/lib/typo3-dummy/execdir/';
+$TYPO3_CONF_VARS['GFX']["im_path_lzw"] = '/var/lib/typo3-dummy/execdir/';
+
+// Set Value to 1 if version of ImageMagick is greater than 4.9
+$TYPO3_CONF_VARS["GFX"]["im_version_5"] = 'gm';
+
+// This variable can be empty if ImageMagick is compiled with LZW.
+// Otherwise you have to set the path to LZW
+//$TYPO3_CONF_VARS["GFX"]["im_path_lzw"] = '';
+
+// Image file formats that should be accepted by Typo3
+$TYPO3_CONF_VARS["GFX"]["imagefile_ext"] = 'gif,jpg,jpeg,tif,bmp,pcx,tga,png,pdf';
+
+$TYPO3_CONF_VARS["GFX"]["im_noFramePrepended"] = '1';
+
+// Enables the preview of images to make the choice more easy
+$TYPO3_CONF_VARS["GFX"]["thumbnails"] = '1';
+
+// Preview of images in png or gif format.
+// Should be the same as "gdlib_png"
+$TYPO3_CONF_VARS["GFX"]["thumbnails_png"] = '1';
+
+// Check freetype quicktest in the basic configuration if text is
+// exceeding the image borders. If yes, you are using Freetype 2 and
+// need to set TTFdpi to 96 dpi
+$TYPO3_CONF_VARS["GFX"]["TTFdpi"] = '96';
+
+// set memory limit to 48 mb.
+// You may want to set this limit higher to get the extension manager working.
+$TYPO3_CONF_VARS['SYS']['setMemoryLimit'] = '64';
+
+// Defines which of these PHP-features to use for various Charset conversing
+// functions in t3lib_cs.
+$TYPO3_CONF_VARS['SYS']['t3lib_cs_convMethod'] = 'iconv';
+
+// Force TYPO3 to use utf-8 in frontend and backend
+$TYPO3_CONF_VARS['BE']['forceCharset'] = 'utf-8';
+
+// Let TYPO3 init mysql to use utf-8 connections
+$TYPO3_CONF_VARS['SYS']['setDBinit'] = 'SET NAMES utf8;';
+
+// The list of file extensions to be considered as images
+$TYPO3_CONF_VARS['GFX']['imagefile_ext'] = 'gif,jpg,jpeg,tif,bmp,pcx,tga,png,swf,pdf,ai';
+
+// Track images generated by imagemagick in the database (prevents double image rendering)
+$TYPO3_CONF_VARS['GFX']['enable_typo3temp_db_tracking'] = '1';
+
+// Set compat level to current version
+$TYPO3_CONF_VARS['SYS']['compat_version'] = '4.5';
+
+// disable the donate now! popup window
+$TYPO3_CONF_VARS['BE']['allowDonateWindow'] = '0';
+
+// define search path for binaries
+$TYPO3_CONF_VARS['SYS']['binPath'] = '/var/lib/typo3-dummy/execdir/';
+
+// disable logging of deprecated functions
+$TYPO3_CONF_VARS['SYS']['enableDeprecationLog'] = '';
+
 ?>
\ No newline at end of file
